### PR TITLE
Deploying to Minikube docs

### PR DESCRIPTION
**Pull request by leozilla**
### What changes were proposed in this pull request?
Documentation howto deploy an application to Minikube.
Refers to: https://github.com/lightbend/cloudflow/issues/967

### Why are the changes needed?
No documentation for this was present and its not trivial to find out how to deploy to Minikube.

### Does this PR introduce any user-facing change?
Yes, its adding Documentation.

### How was this patch tested?
By creating the docs locally and checking it manually.

There is still one thing left which does not work:
  - I tried to add the page to the navigation bar (on the left side) but it does not show up in the locally generated docs. I don't know why this does not work as it looks pretty straight forward to do this.

Input on the documentation is highly appreciated.

### CODE REVIEW

**Comment by leozilla** [comment]
@andreaTP can u please have a look?

File: docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+For more detailed instructions please refer to the official https://minikube.sigs.k8s.io/docs/handbook/pushing/#4-pushing-to-an-in-cluster-using-registry-addon[Pushing images] documentation of Minikube.
+
+Unfortunately Cloudflow does not support changing the `imagePullPolicy` of application images yet (currently fixed to `imagePullPolicy: Always`),
+which is required by other (easier) options for pushing images to Minikube.
+
+== Setup
+
+ * Enable insecure registries in Minikube.
+ * Enable the `registry` Minikube addon.
+ * Configure Docker ot use the insecure registry.
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc:16]:
```suggestion
 * Configure Docker to use the insecure registry.
```

File: docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+You can verify this once Minikube is started by using `minikube ip`.
+
+Start Minikube with insecure registries and enable the `registry` addon:
+
+   minikube start --kubernetes-version=latest --insecure-registry "10.0.0.0/24,192.168.39.0/24"
+   minikube addons enable registry
+
+Configure Docker to use the Minikube registry as insecure registry by adding it to
+the `daemon.json` file, whose default location is `/etc/docker/daemon.json` on Linux.
+If the `daemon.json` file does not exist, create it.
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc:30]: Please add some information on how to do this for Windows / Mac, some details here: https://docs.docker.com/registry/insecure/#deploy-a-plain-http-registry (and probably smart to link to this page)

File: docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+Replace `MINIKUBE_IP` with the actual Minikube VM ip obtained via `Minikube ip`.
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc:32]:
```suggestion
Replace `MINIKUBE_IP` with the actual Minikube VM ip obtained via `minikube ip`.
```

File: docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+You can safely ignore the warnings, those are expected when setting the Docker registry to `None`.
+
+Because we skipped the pushing of the image in the build process we now need to manually push our image to the Minikube registry.
+For this we need to create a tag for the Minikube registry which refers to the original image.
+
+   docker tag localhost:5000/sensor-data-scala:0.1 $(minikube ip):5000/sensor-data-scala:0.1
+   docker push $(minikube ip):5000/sensor-data-scala:0.1
+
+At this point, the Docker image containing the streamlets has been published to the Minikube Docker registry.
+The `sensor-data-scala.json` file that the build produces contains the application descriptor.
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/get-started/pages/deploy-to-k8s-minikube.adoc:90]:
```suggestion
The `sensor-data-scala.json` file that the build produces, contains the application descriptor.
```

**Review by RayRoestenburg — COMMENTED** [review]
Some minor comments and suggestions. otherwise good to go, thanks @leozilla !

**Comment by leozilla** [comment]
@RayRoestenburg thanks for the fast feedback! I just incorporated your suggestions.
Did u also have a look if I configured the nav panel correctly? Cos locally on my machine this does not show the link to the Minikube page.

**Comment by RayRoestenburg** [comment]
You can check the docs with `make local-preview` in `docs` directory, point browser to `http://localhost:8000/docs/dev/index.html`. is shows up like this in nav panel:
image

which looks good to me for now.
The existing 'deploy to kubernetes cluster' is part of the getting started doc. 
It's probably nice to have this info under "Developing Cloudflow Applications", as well outside of the getting started guide, describing deployment aspects in general.

**Review by RayRoestenburg — APPROVED** [review]

**Comment by RayRoestenburg** [comment]
Thanks @leozilla !

**Pull request merged**